debugger/src/debug_adapter.rs

```diff
@@ -1125,41 +1157,219 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
+                            address: format!("{:#010X}", read_pointer),
+                            column: None,
+                            end_column: None,
+                            end_line: None,
+                            instruction: "<instruction address not readable>".to_string(),
+                            instruction_bytes: None,
+                            line: None,
+                            location: None,
+                            symbol: None,
+                        });
+                        continue;
+                    }
+                }
+            }
+
+            match core_data
+                .capstone
+                .disasm_count(&code_buffer, instruction_pointer as u64, 1)
+            {
+                Ok(instructions) => {
+                    if instructions.len().is_zero() {
+                        read_more_bytes = true;
+                        continue;
+                    }
+
+                    let mut result_instruction = instructions
+                        .iter()
+                        .map(|instruction| {
+                            // Before processing, update the code buffer appropriately
+                            code_buffer = code_buffer.split_at(instruction.bytes().len() as usize).1.to_vec();
+                            read_more_bytes = code_buffer.is_empty();
+
+                            // Move the instruction_pointer for the next read.
+                            instruction_pointer += instruction.bytes().len() as u64;
+
+                            // Try to resolve the source location for this instruction.
+                            // If we find one, we use it ONLY if it is different from the previous one (stored_source_location).
+                            // - This helps to reduce visual noise in the VSCode UX, by not displaying the same line of source code multiple times over.
+                            // If we do not find a source location, then just return the raw assembly without file/line/column information.
+                            let mut location = None;
+                            let mut line = None;
+                            let mut column = None;
+                            if let Some(current_source_location) = core_data
+                                .debug_info
+                                .get_source_location(instruction.address()) {
+                                if let Some(previous_source_location) = stored_source_location.clone() {
+                                    if current_source_location != previous_source_location {
+                                        location = get_dap_source(&current_source_location);
+                                        line = current_source_location.line.map(|line| line as i64);
+                                        column = current_source_location.column.map(|col| match col {
+                                            ColumnType::LeftEdge => 0_i64,
```

Comment:
why use an `i64` here? :)

Reply:
This maps directly to `DisassembledInstruction::column` in the MS DAP spec which is an `i64`.

Reply:
Hmm I personally think this is weird and usize would be better fitting. But we can go with that for sure :)